SilKit/source/core/internal/LoggingDatatypesInternal.hpp

```diff
@@ -92,18 +99,54 @@ std::ostream& operator<<(std::ostream& out, const SourceLoc& sourceLoc)
                << "line=" << sourceLoc.line << ", funcname={\"" << sourceLoc.funcname << "\"}";
 }
 
+inline std::string to_string(const std::unordered_map<std::string, std::string>& kv)
+{
+    std::stringstream outStream;
+    outStream << kv;
+    return outStream.str();
+}
+
+
+inline std::ostream& operator<<(std::ostream& out, const std::unordered_map<std::string, std::string>& kv)
+{
+    std::string result;
+    result.reserve(kv.size() * 2);
+
+    if (kv.size() > 0)
+    {
+        result.append(", kv: ");
+
+        std::unordered_map<std::string, std::string>::const_iterator it = kv.begin();
+        result.append("{");
+        while (it != kv.end())
+        {
+            if (it != kv.begin())
+            {
+                result.append(",");
+            }
+            result.append("\"" + it->first + "\"" + ":" + "\""
+                          + it->second + "\"");
+            ++it;
+        }
+        result.append("}");
+    }
+    return out << result;
+}
+
+
 std::string to_string(const LogMsg& msg)
 {
     std::stringstream outStream;
     outStream << msg;
     return outStream.str();
 }
 
+
 std::ostream& operator<<(std::ostream& out, const LogMsg& msg)
 {
-    out << "LogMsg{logger=" << msg.logger_name << ", level=" << msg.level
+    out << "LogMsg{logger=" << msg.loggerName << ", level=" << msg.level
         << ", time=" << msg.time.time_since_epoch().count() << ", source=" << msg.source << ", payload=\""
-        << msg.payload << "\""
+        << msg.payload << msg.keyValues << "\""
```

Comment:
I think the formatting here isn't correct, the map written between the quotes surrounding the `payload` contents.